planner/cascades/transformation_rules.go

```diff
@@ -992,3 +993,34 @@ func (r *MergeAggregationProjection) OnTransform(old *memo.ExprIter) (newExprs [
 	newAggExpr.SetChildren(old.Children[0].GetExpr().Children...)
 	return []*memo.GroupExpr{newAggExpr}, true, false, nil
 }
+
+// MergeAdjacentSelection merge adjacent selection
+type MergeAdjacentSelection struct {
+	baseRule
+}
+
+// NewRuleMergeAdjacentSelection creates a new Transformation MergeAdjacentSelection.
+// The pattern of this rule is `Selection->Selection->X` to `Selection->X`.
+func NewRuleMergeAdjacentSelection() Transformation {
+	rule := &MergeAdjacentSelection{}
+	rule.pattern = memo.BuildPattern(
+		memo.OperandSelection,
+		memo.EngineTiKVOrTiFlash,
+		memo.NewPattern(memo.OperandSelection, memo.EngineTiKVOrTiFlash),
```

Comment:
We can use `EngineAll` here, since all of them support `Selection`.

Reply:
ok, I'll fx it